client/core/core.go

```diff
@@ -1540,13 +1540,37 @@ func (c *Core) Logout() error {
 // initializeDEXConnections connects to the DEX servers in the conns map and
 // authenticates the connection. If registration is incomplete, reFee is run and
 // the connection will be authenticated once the `notifyfee` request is sent.
+// If an account is not found on the dex server upon dex authentication the
+// account is disabled, initialize() will be triggered which will result in the
+// user being prompted to register again.
 func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) []*DEXBrief {
 	// Connections will be attempted in parallel, so we'll need to protect the
 	// errorSet.
 	var wg sync.WaitGroup
 	c.connMtx.RLock()
-	defer c.connMtx.RUnlock()
 	results := make([]*DEXBrief, 0, len(c.conns))
+	var reconcileConnectionsWg sync.WaitGroup
+	reconcileConnectionsWg.Add(1)
+	disabledAccountHostChan := make(chan string)
+	disabledAccountHosts := make(map[string]bool)
```

Comment:
when the value doesn't matter, it's idiomatic to use empty structs. i.e. `map[string]struct{}`

Could also use a `chan *dexConnection` which may simplify just a little.

Reply:
I've simplified things a bit by executing a map delete as you and buck have suggested.